services/ingestion/etl/wifi4eu/xls/src/lib/transform.js

```diff
@@ -0,0 +1,253 @@
+
+/**
+ * Preprocess budget field. For the moment, we don't have this data.
+ *
+ * @memberof Wifi4EuXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {Budget}
+ */
+const getBudget = () => ({
+  total_cost: { value: 0, currency: '', raw: '' },
+  eu_contrib: {
+    value: 0,
+    currency: 'EUR',
+    raw: '',
+  },
+  private_fund: { value: 0, currency: '', raw: '' },
+  public_fund: { value: 0, currency: '', raw: '' },
+  other_contrib: { value: 0, currency: '', raw: '' },
+  funding_area: [],
+  mmf_heading: '',
+});
+
+/**
+ * Preprocess `call_year` field. Data comes from `record['Call year']`.
+ *
+ * @memberof Wifi4EuXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {String}
+ */
+const getCallYear = record => (record['Call year'] ? record['Call year'] : '');
+
+/**
+ * Preprocess `description` field. Data comes from `record['Project Description']`.
+ *
+ * @memberof Wifi4EuXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {String}
+ */
+const getProjectDescription = record =>
+  record['Project Description'] ? record['Project Description'] : '';
+
+/**
+ * Preprocess `partners` field.
+ *
+ * Data comes from the following source fields:
+ * - `municipality name`
+ *
+ * @memberof Wifi4EuXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {String}
+ */
+const getBeneficieries = record => {
+  const name = record['municipality name'] ? record['municipality name'] : '';
+
+  return [
+    {
+      address: '',
+      country: '',
+      email: '',
+      name,
+      phone: '',
+      region: '',
+      role: 'beneficiary',
+      type: '',
+      website: '',
+    },
+  ];
+};
+
+/**
+ * Helper to build an address value.
+ *
+ * Data comes from the following source fields:
+ * - `address`
+ * - `address num`
+ *
+ * @memberof Wifi4EuXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {String} Optimistic concatenation between `address` and `address num`.
+ */
+const getAddress = record => {
+  let address = '';
+
+  const addressLoc = record.address ? record.address : '';
+  const addressNum =
+    record['address num'] !== 's/n' && undefined ? record['address num'] : '';
+
+  // Build address based on available information.
+  if (addressLoc && addressNum) {
+    address = `${addressLoc} ${addressNum}`;
+  } else if (addressLoc) {
+    address = addressLoc;
+  }
+
+  return address;
+};
+
+/**
+ * Generates values for `project_id` field since source data misses these.
+ * It's needed for having separate projects in the Elasticsearch database.
+ *
+ * Depends on `getAddress` helper function.
+ *
+ * @memberof Wifi4EuXlsTransform
+ * @param {Object} record The row received from parsed file
+ * @returns {String}
+ */
+const getProjectId = record =>
+  crypto
+    .createHash('md5')
+    .update(getAddress(record))
```

Comment:
Laurent told us we could use the `municipality name` as a unique identifier because there should not be more than 1 budget allocation per city. If there are multiple entries with the same `municipality name`, they should be considered as duplicates.

That's the theory.

It would be good though to confirm that those entries with the same `municipality name` are actual duplicates or if they differ. (let's check the xls file)